### PR TITLE
Switch profiling to use SyNanosecondsSinceEpoch for timing

### PR DESCRIPTION
**Pull request by ChrisJefferson**
As title, use SyNanosecondsSinceEpoch in profile.c. Addresses #749

### CODE REVIEW

File: src/profile.c [hunk 1]
```diff
@@ -361,23 +357,30 @@ static inline UInt getFilenameId(Stat stat)
   return id;
 }
 
+static inline Int8 CPUmicroseconds()
+{
+#if defined(HAVE_GETTIMEOFDAY)
+  struct timeval timebuf;
```
Comment (markuspf) [inline src/profile.c]: Should you not be checking for `HAVE_GETRUSAGE` here?

Comment (markuspf) [inline src/profile.c]: Just to make sure you didn't miss this. I'd like to merge, but I think `HAVE_GETRUSAGE` is the correct constant to `\#ifdef` here?

**Comment by ChrisJefferson** [comment]
Patch now cleaned up (sorry for the delay)

**Comment by markuspf** [comment]
Thanks!

**Pull request merged**